[docs][web-browser]: update to clarify that iOS uses ASWebAuthenticationSession for auth

sondrateconsulting-ryan opened the pull request:
Follow up to #18320

# Why

<!--
Please describe the motivation for this PR, and link to relevant GitHub issues, forums posts, or feature requests.
-->

# How

<!--
How did you build this feature or fix this bug and why?
-->

# Test Plan

<!--
Please describe how you tested this change and how a reviewer could reproduce your test, especially if this PR does not include automated tests! If possible, please also provide terminal output and/or screenshots demonstrating your test/reproduction.
-->

# Checklist

<!--
Please check the appropriate items below if they apply to your diff. This is required for changes to Expo modules.
-->

- [ ] Documentation is up to date to reflect these changes (eg: https://docs.expo.dev and README.md).
- [ ] Conforms with the [Documentation Writing Style Guide](https://github.com/expo/expo/blob/main/guides/Expo%20Documentation%20Writing%20Style%20Guide.md)
- [ ] This diff will work correctly for `npx expo prebuild` & EAS Build (eg: updated a module plugin).

sondrateconsulting-ryan committed:
[docs][web-browser]: update to clarify that iOS uses ASWebAuthenticationSession for auth

Follow up to #18320

Simek commented:
Hello @sondrateconsulting-ryan, nice catch! 👍 

Can you also updated the versioned SDK doc pages, since this seems to be a case from SDK 46 onwards?
* https://github.com/search?q=repo%3Aexpo%2Fexpo%20SFAuthenticationSession&type=code

sondrateconsulting-ryan committed:
[docs][web-browser]: update to clarify that iOS uses ASWebAuthenticationSession for auth

sondrateconsulting-ryan committed:
[docs][web-browser]: update to clarify that iOS uses ASWebAuthenticationSession for auth

sondrateconsulting-ryan committed:
[docs][web-browser]: update to clarify that iOS uses ASWebAuthenticationSession for auth

sondrateconsulting-ryan commented:
@Simek, updated throughout the repo with the exception of [DevLauncherAuth](https://github.com/expo/expo/blob/4a8405a1640eac18a0b71e153d2ff1c0fbb9b57d/packages/expo-dev-launcher/ios/DevLauncherAuth.swift#L37) where `SFAuthenticationSession` is truly used

I'm not familiar with the repo, so I modified the built files since they're committed. I can revert if necessary, just lmk

sondrateconsulting-ryan committed:
restore built files
